client/state/selectors/is-mapped-domain-site.js

```diff
@@ -26,8 +20,5 @@ export default function isMappedDomainSite( state, siteId ) {
 		return null;
 	}
 
-	return (
-		get( site, 'options.is_mapped_domain', false ) &&
-		some( domains, ( { isWPCOMDomain } ) => ! isWPCOMDomain )
-	);
+	return site.options?.is_mapped_domain && domains.some( ( { isWPCOMDomain } ) => ! isWPCOMDomain );
```

Comment:
I had to fix the unit test here. It has a wrong mock for `sites.domains`. Instead of returning an array, it returned a `{ isWPCOMDomain: true }` object. And `_.some` then iterated over the `true` value. Using `Array.prototype.some` revealed that.

I changed the mock to an array, and the `isWPCOMDomain` is `false` now. Previously, `true.isWPCOMDomain` evaluated to a falsy `undefined` value.